temporal/server.go

```diff
@@ -194,18 +198,32 @@ func (s *Server) Start() error {
 			return err
 		}
 
+		s.serviceStoppedChs[svcName] = make(chan struct{})
+
 		var svc common.Daemon
 		switch svcName {
 		case primitives.FrontendService:
 			svc, err = frontend.NewService(params)
 		case primitives.HistoryService:
-			var historySvc *history.Service
+			// todo: generalize this custom case logic as other services onboard fx
 			histApp := fx.New(
-				fx.Supply(params),
-				history.Module,
-				fx.Populate(&historySvc))
+				fx.Supply(
+					params,
+					s.serviceStoppedChs[svcName],
```

Comment:
this simply supply a channel, how is this wired up? where does it close this channel when service close?

Reply:
Used by lifetime hooks. Currently it's generic, but I want to keep it simple for now and solve collisions further down the line.